### PR TITLE
:sparkles: Reusable button#23

### PR DESCRIPTION
**Pull request by Alexotieno1717**
# Description

Created a reusable normal button component that I can reuse anywhere it's needed across the application

Reference the issue related to this PR as shown below. Every issue has its own number you can find the issue number [here](https://github.com/italanta/elewa-website/issues/23).

## Type of change

- [✅] New feature (non-breaking change which adds functionality)

## Screenshot (optional)
The Website before the changes.

Screenshot from 2023-08-10 10-27-31

The Website after implementing the Reusable Normal button component.

Screenshot from 2023-08-10 09-45-32

## Checklist

- [✅] My code follows the style guidelines of this project
- [✅] I have performed a self-review of my code
- [✅] I have commented my code, particularly in hard-to-understand areas
- [✅] I have made corresponding changes to the documentation

### CODE REVIEW

File: apps/elewa-website/src/app/app.component.ts [hunk 1]
```diff
@@ -7,4 +7,9 @@ import { Component } from '@angular/core';
 })
 export class AppComponent {
   title = 'elewa-website';
+  
+  /* data that will be display when i hover over my button */
+  myButtonData = {
+    text: 'Contact us',
+  };
```
Comment (LemmyMwaura) [inline apps/elewa-website/src/app/app.component.ts:14]: No need to push this, as our app component does not need it.

File: libs/elements/layout/buttons/src/lib/normal-button/normal-button.component.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  * Used to trigger actions.
+  * */
+  @Output() buttonClick: EventEmitter<void> = new EventEmitter<void>();
+
+
+  /* Onclick function emits the event, and any parent component
+   that is listening for the buttonClick event will be notified  */
+  onClick(): void{
+    this.buttonClick.emit();
+  }
```
Comment (LemmyMwaura) [inline libs/elements/layout/buttons/src/lib/normal-button/normal-button.component.ts:27]: Good

File: libs/elements/layout/buttons/src/lib/normal-button/normal-button.component.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import { ButtonData } from "../../../../../../models/schema/ui/buttons/button.interface";
```
Comment (LemmyMwaura) [inline libs/elements/layout/buttons/src/lib/normal-button/normal-button.component.ts:2]: Should use aliased imports. `@elewa...`

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Good first attempt. Just a few changes remaining

**Comment by Alexotieno1717** [comment]
@LemmyMwaura I have made changes to this PR.

File: libs/elements/layout/buttons/src/lib/normal-button/normal-button.component.ts (hunk 3 above)
Comment (Alexotieno1717) [reply libs/elements/layout/buttons/src/lib/normal-button/normal-button.component.ts]: okay noted

File: libs/elements/layout/buttons/src/lib/normal-button/normal-button.component.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  // eslint-disable-next-line @angular-eslint/component-selector
+  selector: 'elewa-normal-button',
+  templateUrl: './normal-button.component.html',
+  styleUrls: ['./normal-button.component.scss'],
+})
+export class NormalButtonComponent {
+  /* Defined the Input decorator
+  * passing data to child component
+  * */
+  @Input() buttonData:ButtonData = {};
```
Comment (LemmyMwaura) [inline libs/elements/layout/buttons/src/lib/normal-button/normal-button.component.ts:14]: Replace with `@Input() buttonData !: ButtonData`

File: libs/elements/layout/buttons/src/lib/normal-button/normal-button.component.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<!-- This is a button element with custom styling and behavior -->
+<button
+  class="reusable-custom-button"
+  [style.backgroundColor]="buttonData.bgColor"
+  [style.color]="buttonData.color"
+  (click)="onClick()"
+  [title]="buttonData.text"
+>
+  <!-- ng-content is allows you to put your own text or content inside the button -->
+  <ng-content></ng-content>
```
Comment (LemmyMwaura) [inline libs/elements/layout/buttons/src/lib/normal-button/normal-button.component.html:10]: Interpolate this text {{ yourProperty }}

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
Just One last thing, kindly resolve this

File: libs/elements/layout/buttons/src/lib/normal-button/normal-button.component.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import { ButtonData } from "@elewa-website/models/schema/ui/buttons";
+
+@Component({
+  // eslint-disable-next-line @angular-eslint/component-selector
+  selector: 'elewa-normal-button',
```
Comment (LemmyMwaura) [inline libs/elements/layout/buttons/src/lib/normal-button/normal-button.component.ts:6]: Remove this and then rename to `elewa-website-normal-button`.

**Pull request closed without merging**